Composer/packages/client/src/components/AddRemoteSkillModal/PreparatoryWorkDialog.tsx

```diff
@@ -0,0 +1,211 @@
+  label: string;
+  description: string;
+  required?: boolean;
+};
+const CustomLabel: React.FC<CustomLabelProps> = ({ label, description, required = false }) => {
+  return (
+    <Stack style={{ display: 'flex', flexDirection: 'row' }}>
+      <Label
+        required={required}
+        styles={{ root: { marginRight: '4px', selectors: { '::after': { paddingRight: '0px' } } } }}
+      >
+        {label}
+      </Label>
+      <TooltipHost content={description}>
+        <FontIcon iconName="Info" style={{ fontSize: FontSizes.size12 }} />
+      </TooltipHost>
+    </Stack>
+  );
+};
+
+const renderMicrosoftAppId = (MicrosoftAppId: string, label: string, description: string): JSX.Element => {
+  return (
+    <Fragment>
+      <CustomLabel description={description} label={label} />
+      <div
+        style={{
+          color: NeutralColors.gray90,
+          backgroundColor: NeutralColors.gray20,
+          fontSize: FontSizes.size14,
+          display: 'flex',
+          justifyContent: 'space-between',
+          lineHeight: '30px',
+          paddingLeft: '10px',
+          width: '394px',
+        }}
+      >
+        {MicrosoftAppId}
+        <IconButton
+          iconProps={{ iconName: 'copy' }}
+          styles={{ icon: { fontSize: FontSizes.size12, color: CommunicationColors.primary } }}
+          onClick={() => navigator.clipboard.writeText(MicrosoftAppId)}
+        />
+      </div>
+    </Fragment>
+  );
+};
+export const PreparatoryWorkDialog: React.FC<PreparatoryWorkDialogProps> = (props) => {
+  const { projectId, onDismiss, onNext, onGotoCreateProfile } = props;
+  const settings = useRecoilValue(settingsState(projectId));
+  const botName = useRecoilValue(botDisplayNameState(projectId));
+  const { publishTargets, MicrosoftAppId } = settings;
+  const publishTargetOptions: IDropdownOption[] = publishTargets.map((target) => ({
+    key: target.name,
+    text: target.name,
+  }));
+
+  const [currentTargetName, setCurrentTargetName] = useState(publishTargets.length === 0 ? '' : publishTargets[0].name);
+
+  return (
+    <Fragment>
+      {publishTargets.length === 0 ? (
+        <Fragment>
+          {renderItem(getCreateProfileDescription(botName, onGotoCreateProfile))}
+          {renderItem(appIdInfo)}
+          {renderItem(manifestUrl)}
+        </Fragment>
+      ) : (
+        <Fragment>
+          {renderItem(appIdInfo)}
+          {publishTargets.length === 1 ? (
+            <div style={{ margin: '0 0 40px 30px' }}>
+              {renderMicrosoftAppId(
+                MicrosoftAppId,
+                formatMessage('Your bot’s Microsoft App Id'),
+                formatMessage('Microsoft App Id')
+              )}
+            </div>
+          ) : (
+            <div style={{ display: 'flex', margin: '0 0 40px 30px' }}>
+              <div style={{ marginRight: '20px' }}>
+                <CustomLabel
+                  required
+                  description={formatMessage('Publish profile')}
```

Comment:
if the description and label are the same, what's the point of having both?

Reply:
The desciption is what i set, because the design dont have it. I think may be the descption is more detail than label, so i use two params